### PR TITLE
Use embed_weights instead of vocab.vectors

### PR DESCRIPTION
**Pull request by Eleven1Liu**
**Description** Use the original `embed_weights` to avoid floating point error while reading embedding from file.

1. `torch.Tensor(x).numpy()` is not x. Please find more info here:
https://github.com/pytorch/text/blob/a27bafe2bb0928dbcf66b2fb3a59d132960ca955/torchtext/vocab.py#L216

2. This issue affect the reproducibility of `VanillaCNN` in the `reproduce` branch.

### CODE REVIEW

File: libmultilabel/data_utils.py [hunk 1]
```diff
@@ -165,20 +164,21 @@ def load_or_build_text_dict(
...
+        vocabs.set_vectors(vocabs.stoi, torch.Tensor(embedding_weights), dim=dim)
     elif not embed_file.isdigit():
         logging.info(f'Load pretrained embedding from torchtext.')
         vocabs.load_vectors(embed_file, cache=embed_cache_dir)
     else:
         raise NotImplementedError
 
     if normalize_embed:
-        embedding_weights = copy.deepcopy(vocabs.vectors.detach().cpu().numpy())
+        embedding_weights = embedding_weights if embedding_weights else vocabs.vectors.numpy()
```
Comment (cjlin1) [inline libmultilabel/data_utils.py:176]: a short comment here may be useful. Otherwise people ask why we want to do the if statement.

File: libmultilabel/data_utils.py [hunk 2]
```diff
@@ -165,20 +164,23 @@ def load_or_build_text_dict(
...
     elif not embed_file.isdigit():
         logging.info(f'Load pretrained embedding from torchtext.')
         vocabs.load_vectors(embed_file, cache=embed_cache_dir)
     else:
         raise NotImplementedError
 
     if normalize_embed:
-        embedding_weights = copy.deepcopy(vocabs.vectors.detach().cpu().numpy())
+        # Torchtext convert vectors to `torch.Tensor()`(float32) which makes floating point error after the
+        # conversion between tensor and numpy. We use the original `embedding_weights` here for consistency.
```
Comment (cjlin1) [inline libmultilabel/data_utils.py:177]: still have some questions on this. Let's discuss it on Th.

File: libmultilabel/data_utils.py [hunk 3]
```diff
@@ -165,20 +164,23 @@ def load_or_build_text_dict(
...
+        vocabs.set_vectors(vocabs.stoi, torch.Tensor(embedding_weights), dim=dim)
     elif not embed_file.isdigit():
         logging.info(f'Load pretrained embedding from torchtext.')
         vocabs.load_vectors(embed_file, cache=embed_cache_dir)
     else:
         raise NotImplementedError
 
     if normalize_embed:
-        embedding_weights = copy.deepcopy(vocabs.vectors.detach().cpu().numpy())
+        # `embedding_weights` is converted to torch.FloatTensor in `vocabs.set_vectors`.
```
Comment (cjlin1) [inline libmultilabel/data_utils.py:176]: how about:

vocabs.vectors is a torch.FloatTensor from the result of `vocabs.set_vectors` earlier. To have better precision for calculating the normalization, we use the original `embedding_weights,` a torch.DobleTensor, if it is available.

Comment (Eleven1Liu) [reply libmultilabel/data_utils.py:176]: Great a second!

**Review by cjlin1 — APPROVED** [review]
